lxd/instance/drivers/driver_qemu.go

```diff
@@ -3853,6 +3857,41 @@ func (d *qemu) addNetDevConfig(busName string, qemuDev map[string]string, bootIn
 				return fmt.Errorf("Failed setting up device %q: %w", devName, err)
 			}
 
+			return nil
+		}
+	} else if shared.PathExists(vhostVDPAPath) {
+		monHook = func(m *qmp.Monitor) error {
+			reverter := revert.New()
+			defer reverter.Fail()
+
+			err := os.Chown(vhostVDPAPath, int(d.state.OS.UnprivUID), -1)
+			if err != nil {
+				return fmt.Errorf("Failed to chown vDPA device %q: %w", vhostVDPAPath, err)
+			}
+
+			qemuNetDev := map[string]any{
+				"id":       fmt.Sprintf("vhost-%s", vDPADevName),
+				"type":     "vhost-vdpa",
+				"vhostdev": vhostVDPAPath,
```

Comment:
We need to change this to use `vhostfd` like you had it earlier (i.e using `monitor.SendFile()` like `macvlan`).

Reply:
yep testing it right now